lib/ca_testing/drivers/v4/browserstack.rb

```diff
@@ -1,6 +1,113 @@
 # frozen_string_literal: true
 
-require "ca_testing/drivers/v4/browserstack/android"
-require "ca_testing/drivers/v4/browserstack/base"
-require "ca_testing/drivers/v4/browserstack/chrome"
-require "ca_testing/drivers/v4/browserstack/ios"
+require "faraday"
+
+require "ca_testing/drivers/v4/capabilities"
+require "ca_testing/drivers/v4/options"
+
+module CaTesting
+  module Drivers
+    module V4
+      class Browserstack
+        attr_reader :browser, :browserstack_options, :device_options
+        private :browser, :browserstack_options, :device_options
+
+        def initialize(browser, browserstack_options, device_options = {})
+          @browser = browser
+          @browserstack_options = browserstack_options
+          @device_options = device_options
+        end
+
+        # @return [Nil]
+        #
+        # Register a new driver with the default selenium name for use in a remote browserstack setup
+        def register
+          Capybara.register_driver :selenium do |app|
+            Capybara::Selenium::Driver.new(
+              app,
+              browser: :remote,
+              capabilities: [desired_capabilities, options],
+              url: browserstack_hub_url
+            )
+          end
+        end
+
+        private
+
+        def desired_capabilities
+          Selenium::WebDriver::Remote::Capabilities.new(
+            ::Faraday::Utils.deep_merge(
```

Comment:
Is Faraday included just for this?

Reply:
Yeh basically. However down the line when I migrate more there will be the exit handler registration in here